isisd/isis_tlvs.c

```diff
@@ -289,11 +304,6 @@ static void format_item_ext_subtlvs(struct isis_ext_subtlvs *exts,
 
 		for (adj = (struct isis_adj_sid *)exts->adj_sid.head; adj;
 		     adj = adj->next) {
-			if (((mtid == ISIS_MT_IPV4_UNICAST)
```

Comment:
Well, when you setup only IPv6 address (or only IPv4) without Multi-Topology enable, the Adjacency SID is not advertised because the test is too strict. It is also the same if you setup both IPv6 and IPv4 but without Multi-Topology enable.

Thus I removed the test to be sure that Adjacency SID will be correctly advertised in any scenario (with or without Multi-Topology, with single IPv6 or IPv4 address or with both IPv4 & IPv6 addresses)